File: backend/internal/model/epic.go
package model

import "time"

type Epic struct {
	EpicID      uint      `gorm:"primaryKey" json:"epicId"`
	ProjectID   uint       `gorm:"index;not null" json:"projectId"`
	Title       string     `gorm:"not null" json:"title"`
	Description string     `json:"description"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`
}

File: frontend/src/types/epic.ts
// epic.ts
export interface Epic {
  epicId: number;
  projectId: number;
  title: string;
  description: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateEpicPayload {
  title: string;
  description?: string;
}

export interface UpdateEpicPayload {
  title?: string;
  description?: string;
}